gazelle/buf/buf_lock.go

```diff
@@ -0,0 +1,22 @@
+package buf
+
+import "os"
+
+type bufLock struct {
+	Version string `yaml:"version,omitempty" json:"version,omitempty"`
+	Deps    []struct {
+		Remote     string `yaml:"remote,omitempty" json:"remote,omitempty"`
+		Owner      string `yaml:"owner,omitempty" json:"owner,omitempty"`
+		Repository string `yaml:"repository,omitempty" json:"repository,omitempty"`
+		Commit     string `yaml:"commit,omitempty" json:"commit,omitempty"`
+	} `yaml:"deps,omitempty" json:"deps,omitempty"`
+}
+
+func readBufLock(file string) (*bufLock, error) {
```

Comment:
Nit: `readBufWork` and `readBufLock` seem almost identical and are both only used in one occasion. I'd remove these helpers and just inline the `os.ReadFile`/`parseJsonOrYaml` rather than adding the layer of indirection.